tests/test_component.py

```diff
@@ -373,6 +373,16 @@ def test_get_component_by_id_raises_on_missing_component(self):
         with pytest.raises(KeyError):
             get_component_by_class_id("nonexistent")
 
+    def test_component_render_id(self):
```

Comment:
Unrelated to this PR, but I realized there wasn't a test to check the accessing of `Component.id`.